src/TypeParser/JMSTypeParser.php

```diff
@@ -65,12 +66,13 @@ private function parseType(array $typeInfo, bool $isSubType = false): PropertyTy
             return new PropertyTypeClass($typeInfo['name'], $nullable);
         }
 
-        if (self::TYPE_ARRAY === $typeInfo['name']) {
+        $isCollection = self::TYPE_ARRAY_COLLECTION === $typeInfo['name'];
```

Comment:
couldn't people also be using `Doctrine\Common\Collections\ArrayCollection` in their annotation?

could we respect type imports and do an is_subclass_of check?

Reply:
Fixed and adjusted via 50d0172cea019bfc227eacd8aacebc4911a31d61 - I had to still support `ArrayCollection` as this is a keyword that is supported in the in the [ArrayCollectionHandler from JMS](https://github.com/schmittjoh/serializer/blob/master/src/Handler/ArrayCollectionHandler.php#L23)